### PR TITLE
Fix docs

### PR DESCRIPTION
**Pull request by LouiseKlodt**
# Summary of changes
Fixes the code examples and documentation for `test-machine`

- [ ] tests
- [ ] updated [CHANGELOG](../CHANGELOG.md) (the "unreleased" section)

### CODE REVIEW

**Comment by cddr** [comment]
Thanks for giving the docs some :heart: Louise. Particularly the note about `:partition-count`. That's a common stumbling block.

**Review by minimal — APPROVED** [review]

File: doc/test-machine.md [hunk 1]
```diff
@@ -31,32 +31,42 @@ to these services in a shared environment like uat/staging.
 (ns my.app-test
   (:require
     [my.app :as app]
-    [jackdaw.test :as j.t]
+    [jackdaw.serdes :refer [string-serde edn-serde]]
+    [jackdaw.serdes.json :as json-serde]
```
Comment (gphilipp) [inline doc/test-machine.md:35]: I'm not a fan of the ns alias name, it's easily confused with a function name (like the 2 above). Can you name it `serdes.json` or even `jsj`

File: doc/test-machine.md [hunk 2]
```diff
@@ -135,11 +146,43 @@ better that you write this macro yourself so that you can tailor it to your own
...
+  [f transport]
+  (with-fixtures [(topic-fixture kafka-config input-topic-config)
+                  (topic-fixture kafka-config output-topic-config)
+                  (service-ready? {:http-url "http://localhost:8082"
+                                   :timeout 5000})]
+    (with-open [machine (test-machine transport)]
       (f machine))))
 ```
+
+The `topic-fixture` function creates all the topics named in the supplied `topic-config` before running tests. You have to import `TopologyTestDriver` for `test-machine` to work, which requires you to bring in the `org.apache.kafka/kafka-streams-test-utils` library as a dependency, using a version within `2.0.0` - `2.3.0`. The `topic-fixture` expects the `topic-config` to contain `:partition-count` and `:replication-factor` to be present, besides the `topic-name` and key-value serdes.
```
Comment (gphilipp) [inline doc/test-machine.md:188]: Why not above than `2.3.0`?

**Review by gphilipp — CHANGES_REQUESTED** [review]

File: doc/test-machine.md (hunk 1 above)
Comment (LouiseKlodt) [reply doc/test-machine.md:35]: OK, makes sense, thanks!

File: doc/test-machine.md (hunk 2 above)
Comment (LouiseKlodt) [reply doc/test-machine.md:188]: Thanks @gphilipp  - Currently, when bringing in versions of `kafka-streams-test-utils` higher than `2.3.0`,  it throws errors. For example, for `2.4.0`:
```
Syntax error (ClassNotFoundException) compiling new at (jackdaw/test.clj:250:5).
org.apache.kafka.streams.internals.QuietStreamsConfig
```
I assume we need to bring Jackdaw up to date with the newest versions from Apache Kafka's libraries.

I tried all version numbers from `2.4.0, 2.5.0` ... up to the latest `2.8.0` - they all throw errors in `jackdaw/test.clj`.

I tested this in a dummy example app, you can find it here: 
https://github.com/FundingCircle/jackdaw-examples

Comment (gphilipp) [reply doc/test-machine.md:188]: Ok then, let's file a separate issue for that then 👍 .

**Review by gphilipp — APPROVED** [review]

File: doc/test-machine.md (hunk 2 above)
Comment (LouiseKlodt) [reply doc/test-machine.md:188]: OK, opened an issue here: https://github.com/FundingCircle/jackdaw/issues/290

**Pull request merged**